### DISCUSSION TITLE
Zone Forecasts and/or Weather Radio type forecasts

### DISCUSSION THREAD

**Discussion by fury88** [weather-gov/api#639] [Q&A] [answered]
Hi all,
I'm working on a home project using weather api to mimic the NOAA Weather Radio. Do we use Zone to get that particular forecast? I picked coordinates on the map and tried the points forecast but its a little too pinpointed to the area. I'm looking for exactly what is broadcast by NOAA.

Is this available?

-Matt

**Comment by sullynole** [comment, marked as answer]
The NWR forecast is customized for each transmitter based on the zones that station covers in that office's zone forecast text product (ZFP), which is formatted at each office based on the grids.  I am not sure if it can be replicated exactly from the text product since there is software at each office to post-process the ZFP for NWR formatting, but the ZFP text product probably is the closest place to start within the API.  https://api.weather.gov/products/types/ZFP

**Reply by fury88** [reply]
Nice. Thank you! It's too bad they put it all in one long string in the product text haha. Python should handle it well. 
I swear awhile back they had all the pieces of the NWS radio broadcast within the API like Synopsis, climate summary, etc. I'm still digging through trying to find the best options. I know at one point they kept up with the mp3 recordings of this but, at least in my area, the links haven't been updated and they haven't written me back about whether its no longer supported or not. Granted it does say its "experimental."

**Reply by fury88** [reply]
And just as I hit enter I jumped up one level in the API and see a huge list of all the products. Nice! Last time I dealt with the API it was the old XML version. JSON type so much nicer.

**Reply by sullynole** [reply]
The RWR text product is typically used for current observations, but when it gets to actual alerts for weather and other hazards that will get very complicated as it is very dynamic.  

An NWR transmitter has an entire list of text products that it can be configured to play and that is curated at the local office level.  It also has configurations for how and when to play each product when issued.  Offices can even record audio specifically for an NWR transmitter in certain situations.  It will not be possible to mimic exactly what is playing currently on a transmitter.  

I am curious about the use case for recreating NWR on the web from API.   API is designed around the idea that the alerts, forecast, and observations can be tailored to a specific point as opposed to an entire zone so it is much more customized than what would be played on NWR for a wider audience.   I have seen where some individuals have set up a stream of specific transmitters for the web though.

**Reply by fury88** [reply]
I'm beginning to see that now and it matches what you are saying. You're question is a fair one!

My use case is part for fun and part for a non-profit type one.
As you mentioned, I actually DO currently have a setup on RaspberryPi that rebroadcasts the local NWR online. There was a hole in the area and I really love the fact that I can listen to it online so I set up an SDR and am sending it over to Broadcastify. Unfortunately, despite doing everything I can short of mounting a massive antenna on my roof I can't get a signal without SOME static. Frustrating to say the least because my portable HAM and home NWR's tune in fine with no issues. 
So I'm playing around with an experimental TTS version of it using the text from the API. It's actually working rather well and found a voice package that's pretty solid sounding. Anyway, I'd like to offer additional weather services  from my site including the rebroadcast in the area. :-)

**Reply by sullynole** [reply]
If it helps, the periods for a specific zone from the ZFP can also be accessed through the /zones/forecast/{zoneId}/forecast endpoint with additional JSON formatting.

**Reply by fury88** [reply]
Thanks! Actually, that's exactly what I'm looking for because the forecast is a broader range due to a larger grid area (or zone I guess). I'm already parsing something similar with the forecast based on grid coordinates. That part is easy. It's too bad the additional products are as nicely defined as this zone forecast is.
